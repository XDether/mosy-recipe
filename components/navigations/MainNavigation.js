import React, { useEffect, useState } from "react";
import { createNativeStackNavigator } from '@react-navigation/native-stack';
import { DefaultTheme, NavigationContainer } from '@react-navigation/native';
import AddRecipeOverview from "../pages/AddRecipeOverview";
import AddRecipeEditMode from "../pages/AddRecipeEditMode";
import TabNavigation from  "./TabNavigation";
import RecipePage from "../pages/RecipePage";
import Recipe from "../models/Recipe";
import storage from "../helpers/Storage";
import colors from "../constants/colors";
import { TouchableOpacity } from "react-native";
import Ionicons from "@expo/vector-icons/Ionicons";
import { useNavigation } from "@react-navigation/native";


export default function MainNavigation ()
{
  const [tempRecipe, setTempRecipe] = useState({link:""})
  const [dataSet, setDataSet] = useState([]);
  const Stack = createNativeStackNavigator();


  useEffect(() => {
  }, []);

  const navigation = useNavigation();
  const BackButton = () =>{
    return(
    <TouchableOpacity 
      style={{marginRight: 10}}
      onPress={()=>{navigation.navigate("TabScreen")}}
    >
      <Ionicons name="arrow-back-outline" size={24}/>
    </TouchableOpacity>
    )
  }

  return(

      <Stack.Navigator initialRouteName='EditPages' screenOptions={{headerLeft: BackButton}}>

          <Stack.Screen name = "TabScreen" options={{headerShown : false}}>
            {(props) => <TabNavigation {...props} dataSet={dataSet}/> }
          </Stack.Screen>

          <Stack.Screen name="AddRecipeOverview" options={{ headerStyle:{backgroundColor: colors.primary}, title: "Neues Rezept"}}>
            {(props) => <AddRecipeOverview {...props} setTempRecipe={ setTempRecipe } />}
          </Stack.Screen>

          <Stack.Screen name="AddRecipeEditMode" options={{ headerStyle:{backgroundColor: colors.primary}, headerShown: false}}>
            {(props) => <AddRecipeEditMode {...props} tempRecipe={tempRecipe} setTempRecipe = {setTempRecipe} />}
          </Stack.Screen>

          <Stack.Screen name="RecipePage" options={{ headerStyle:{backgroundColor: colors.primary}, title:"Rezept"}}>
            {(props) => <RecipePage {...props}/>}
          </Stack.Screen>
      </Stack.Navigator>)
}