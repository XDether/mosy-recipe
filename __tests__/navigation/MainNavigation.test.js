import React from 'react';
import renderer from 'react-test-renderer'

import MainNavigation from '../../components/navigations/MainNavigation';

it('renders correctly', () => {
  const tree = renderer.create(<MainNavigation />).toJSON();
  expect(tree).toMatchSnapshot();
});
