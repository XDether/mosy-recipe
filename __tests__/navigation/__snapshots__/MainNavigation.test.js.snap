// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNSScreenStack
    style={
      {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      collapsable={false}
      forwardedRef={[Function]}
      gestureResponseDistance={
        {
          "bottom": -1,
          "end": -1,
          "start": -1,
          "top": -1,
        }
      }
      nativeBackButtonDismissalEnabled={false}
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onHeaderBackButtonClicked={[Function]}
      onNativeDismissCancelled={[Function]}
      onTransitionProgress={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      stackPresentation="push"
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      swipeDirection="horizontal"
    >
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backgroundColor="rgb(255, 255, 255)"
        color="rgb(0, 122, 255)"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={true}
        hideBackButton={false}
        largeTitleHideShadow={false}
        title="TabScreen"
        titleColor="rgb(28, 28, 30)"
        topInsetEnabled={false}
        translucent={false}
      />
      <View
        accessibilityElementsHidden={false}
        importantForAccessibility="auto"
        style={
          {
            "flex": 1,
            "flexDirection": "column-reverse",
          }
        }
      >
        <View
          style={
            [
              {
                "flex": 1,
              },
              {
                "backgroundColor": "rgb(242, 242, 242)",
              },
              undefined,
            ]
          }
        >
          <View
            style={
              [
                {
                  "flex": 1,
                },
                undefined,
              ]
            }
          >
            <RNSScreenNavigationContainer
              style={
                {
                  "flex": 1,
                  "overflow": "hidden",
                }
              }
            >
              <RNSScreen
                activityState={2}
                collapsable={false}
                forwardedRef={[Function]}
                gestureResponseDistance={
                  {
                    "bottom": -1,
                    "end": -1,
                    "start": -1,
                    "top": -1,
                  }
                }
                style={
                  {
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                    "zIndex": 0,
                  }
                }
              >
                <View
                  accessibilityElementsHidden={false}
                  importantForAccessibility="auto"
                  style={
                    [
                      {
                        "backgroundColor": "rgb(242, 242, 242)",
                        "flex": 1,
                      },
                      [
                        {
                          "flex": 1,
                          "flexDirection": "column-reverse",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  <View
                    style={
                      {
                        "flex": 1,
                      }
                    }
                  >
                    <View
                      style={
                        {
                          "alignItems": "center",
                          "flex": 1,
                          "justifyContent": "center",
                        }
                      }
                    >
                      <Text>
                        HomeScreen
                      </Text>
                    </View>
                  </View>
                  <View
                    onLayout={[Function]}
                    style={null}
                  >
                    <View
                      collapsable={false}
                      pointerEvents="box-none"
                      style={
                        {
                          "bottom": 0,
                          "left": 0,
                          "position": "absolute",
                          "right": 0,
                          "top": 0,
                          "zIndex": 0,
                        }
                      }
                    >
                      <View
                        collapsable={false}
                        style={
                          {
                            "backgroundColor": "#CE88CB",
                            "borderBottomColor": "rgb(216, 216, 216)",
                            "flex": 1,
                            "shadowColor": "rgb(216, 216, 216)",
                            "shadowOffset": {
                              "height": 0.5,
                              "width": 0,
                            },
                            "shadowOpacity": 0.85,
                            "shadowRadius": 0,
                          }
                        }
                      />
                    </View>
                    <View
                      collapsable={false}
                      pointerEvents="box-none"
                      style={
                        {
                          "height": 44,
                          "maxHeight": undefined,
                          "minHeight": undefined,
                          "opacity": undefined,
                          "transform": undefined,
                        }
                      }
                    >
                      <View
                        pointerEvents="none"
                        style={
                          {
                            "height": 0,
                          }
                        }
                      />
                      <View
                        pointerEvents="box-none"
                        style={
                          {
                            "alignItems": "stretch",
                            "flex": 1,
                            "flexDirection": "row",
                          }
                        }
                      >
                        <View
                          collapsable={false}
                          pointerEvents="box-none"
                          style={
                            {
                              "alignItems": "flex-start",
                              "flexBasis": 0,
                              "flexGrow": 1,
                              "justifyContent": "center",
                              "marginStart": 0,
                            }
                          }
                        />
                        <View
                          collapsable={false}
                          pointerEvents="box-none"
                          style={
                            {
                              "justifyContent": "center",
                              "marginHorizontal": 16,
                              "maxWidth": 718,
                            }
                          }
                        >
                          <Text
                            accessibilityRole="header"
                            aria-level="1"
                            collapsable={false}
                            numberOfLines={1}
                            style={
                              {
                                "color": "rgb(28, 28, 30)",
                                "fontSize": 17,
                                "fontWeight": "600",
                              }
                            }
                          >
                            Home
                          </Text>
                        </View>
                        <View
                          collapsable={false}
                          pointerEvents="box-none"
                          style={
                            {
                              "alignItems": "flex-end",
                              "flexBasis": 0,
                              "flexGrow": 1,
                              "justifyContent": "center",
                              "marginEnd": 0,
                            }
                          }
                        >
                          <View
                            style={
                              [
                                {
                                  "borderRadius": 3,
                                  "overflow": "hidden",
                                },
                                {
                                  "borderRadius": 3,
                                },
                                {
                                  "marginRight": 10,
                                },
                                false,
                              ]
                            }
                          >
                            <View
                              accessibilityRole="button"
                              accessibilityState={
                                {
                                  "busy": false,
                                  "checked": undefined,
                                  "disabled": false,
                                  "expanded": undefined,
                                  "selected": undefined,
                                }
                              }
                              accessibilityValue={
                                {
                                  "max": undefined,
                                  "min": undefined,
                                  "now": undefined,
                                  "text": undefined,
                                }
                              }
                              accessible={true}
                              collapsable={false}
                              focusable={true}
                              onClick={[Function]}
                              onResponderGrant={[Function]}
                              onResponderMove={[Function]}
                              onResponderRelease={[Function]}
                              onResponderTerminate={[Function]}
                              onResponderTerminationRequest={[Function]}
                              onStartShouldSetResponder={[Function]}
                              style={
                                {
                                  "opacity": 1,
                                }
                              }
                            >
                              <View
                                style={
                                  {
                                    "alignItems": "center",
                                    "backgroundColor": "transparent",
                                    "borderColor": "#2089dc",
                                    "borderRadius": 3,
                                    "borderWidth": 0,
                                    "flexDirection": "row",
                                    "justifyContent": "center",
                                    "padding": 8,
                                  }
                                }
                              >
                                <Text />
                              </View>
                            </View>
                          </View>
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </RNSScreen>
            </RNSScreenNavigationContainer>
            <View
              collapsable={false}
              onLayout={[Function]}
              pointerEvents="auto"
              style={
                {
                  "backgroundColor": "#CE88CB",
                  "borderRadius": 100,
                  "borderTopColor": "rgb(216, 216, 216)",
                  "borderTopWidth": 0.5,
                  "bottom": 0,
                  "elevation": 8,
                  "height": 49,
                  "left": 0,
                  "marginBottom": 10,
                  "marginHorizontal": 20,
                  "paddingBottom": 0,
                  "paddingHorizontal": 0,
                  "position": null,
                  "right": 0,
                  "transform": [
                    {
                      "translateY": 0,
                    },
                  ],
                }
              }
            >
              <View
                pointerEvents="none"
                style={
                  {
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
              />
              <View
                accessibilityRole="tablist"
                style={
                  {
                    "flex": 1,
                    "flexDirection": "row",
                  }
                }
              >
                <View
                  accessibilityLabel="Home, tab, 1 of 3"
                  accessibilityRole="button"
                  accessibilityState={
                    {
                      "busy": undefined,
                      "checked": undefined,
                      "disabled": undefined,
                      "expanded": undefined,
                      "selected": true,
                    }
                  }
                  accessibilityStates={
                    [
                      "selected",
                    ]
                  }
                  accessibilityValue={
                    {
                      "max": undefined,
                      "min": undefined,
                      "now": undefined,
                      "text": undefined,
                    }
                  }
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    [
                      {
                        "alignItems": "center",
                        "flex": 1,
                      },
                      {
                        "backgroundColor": "#865DFF",
                      },
                      {
                        "flexDirection": "column",
                        "justifyContent": "flex-end",
                      },
                      {
                        "borderRadius": 100,
                      },
                    ]
                  }
                >
                  <View
                    style={
                      [
                        {
                          "flex": 1,
                        },
                        undefined,
                      ]
                    }
                  >
                    <View
                      style={
                        [
                          {
                            "alignItems": "center",
                            "alignSelf": "center",
                            "height": "100%",
                            "justifyContent": "center",
                            "minWidth": 25,
                            "position": "absolute",
                            "width": "100%",
                          },
                          {
                            "opacity": 1,
                          },
                        ]
                      }
                    >
                      <Text />
                    </View>
                    <View
                      style={
                        [
                          {
                            "alignItems": "center",
                            "alignSelf": "center",
                            "height": "100%",
                            "justifyContent": "center",
                            "minWidth": 25,
                            "position": "absolute",
                            "width": "100%",
                          },
                          {
                            "opacity": 0,
                          },
                        ]
                      }
                    >
                      <Text />
                    </View>
                  </View>
                </View>
                <View
                  accessibilityLabel="Recipes, tab, 2 of 3"
                  accessibilityRole="button"
                  accessibilityState={
                    {
                      "busy": undefined,
                      "checked": undefined,
                      "disabled": undefined,
                      "expanded": undefined,
                      "selected": false,
                    }
                  }
                  accessibilityStates={[]}
                  accessibilityValue={
                    {
                      "max": undefined,
                      "min": undefined,
                      "now": undefined,
                      "text": undefined,
                    }
                  }
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    [
                      {
                        "alignItems": "center",
                        "flex": 1,
                      },
                      {
                        "backgroundColor": "transparent",
                      },
                      {
                        "flexDirection": "column",
                        "justifyContent": "flex-end",
                      },
                      {
                        "borderRadius": 100,
                      },
                    ]
                  }
                >
                  <View
                    style={
                      [
                        {
                          "flex": 1,
                        },
                        undefined,
                      ]
                    }
                  >
                    <View
                      style={
                        [
                          {
                            "alignItems": "center",
                            "alignSelf": "center",
                            "height": "100%",
                            "justifyContent": "center",
                            "minWidth": 25,
                            "position": "absolute",
                            "width": "100%",
                          },
                          {
                            "opacity": 0,
                          },
                        ]
                      }
                    >
                      <Text />
                    </View>
                    <View
                      style={
                        [
                          {
                            "alignItems": "center",
                            "alignSelf": "center",
                            "height": "100%",
                            "justifyContent": "center",
                            "minWidth": 25,
                            "position": "absolute",
                            "width": "100%",
                          },
                          {
                            "opacity": 1,
                          },
                        ]
                      }
                    >
                      <Text />
                    </View>
                  </View>
                </View>
                <View
                  accessibilityLabel="Categorys, tab, 3 of 3"
                  accessibilityRole="button"
                  accessibilityState={
                    {
                      "busy": undefined,
                      "checked": undefined,
                      "disabled": undefined,
                      "expanded": undefined,
                      "selected": false,
                    }
                  }
                  accessibilityStates={[]}
                  accessibilityValue={
                    {
                      "max": undefined,
                      "min": undefined,
                      "now": undefined,
                      "text": undefined,
                    }
                  }
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    [
                      {
                        "alignItems": "center",
                        "flex": 1,
                      },
                      {
                        "backgroundColor": "transparent",
                      },
                      {
                        "flexDirection": "column",
                        "justifyContent": "flex-end",
                      },
                      {
                        "borderRadius": 100,
                      },
                    ]
                  }
                >
                  <View
                    style={
                      [
                        {
                          "flex": 1,
                        },
                        undefined,
                      ]
                    }
                  >
                    <View
                      style={
                        [
                          {
                            "alignItems": "center",
                            "alignSelf": "center",
                            "height": "100%",
                            "justifyContent": "center",
                            "minWidth": 25,
                            "position": "absolute",
                            "width": "100%",
                          },
                          {
                            "opacity": 0,
                          },
                        ]
                      }
                    >
                      <Text />
                    </View>
                    <View
                      style={
                        [
                          {
                            "alignItems": "center",
                            "alignSelf": "center",
                            "height": "100%",
                            "justifyContent": "center",
                            "minWidth": 25,
                            "position": "absolute",
                            "width": "100%",
                          },
                          {
                            "opacity": 1,
                          },
                        ]
                      }
                    >
                      <Text />
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;
